src/utils.ts

```diff
@@ -79,27 +79,55 @@ export function isTrueJoinEvent(event: any): boolean {
  * @param userIdOrGlob A mxid or a glob which is applied to the whole sender field of events in the room, which will be redacted if they match.
  * See `MatrixGlob` in matrix-bot-sdk.
  * @param targetRoomIds Rooms to redact the messages from.
+ * @param isAdmin whether the bot is server admin
  * @param limit The number of messages to redact from most recent first. If the limit is reached then no further messages will be redacted.
  * @param noop Whether to operate in noop mode.
  */
-export async function redactUserMessagesIn(client: MatrixSendClient, managementRoom: ManagementRoomOutput, userIdOrGlob: string, targetRoomIds: string[], limit = 1000, noop = false) {
-    for (const targetRoomId of targetRoomIds) {
-        await managementRoom.logMessage(LogLevel.DEBUG, "utils#redactUserMessagesIn", `Fetching sent messages for ${userIdOrGlob} in ${targetRoomId} to redact...`, targetRoomId);
+export async function redactUserMessagesIn(client: MatrixSendClient, managementRoom: ManagementRoomOutput, userIdOrGlob: string, targetRoomIds: string[], isAdmin: boolean, limit = 1000, noop = false) {
+    const usingGlob = userIdOrGlob.includes("*");
+
+    async function adminRedaction(c: MatrixSendClient, mr: ManagementRoomOutput, userId: string, targetRooms: string[], lim = 1000) {
```

Comment:
typically functions would be pulled out to the top level of the file instead of declared inline. We shouldn't `export` that function though, as it's internal to the file.

I'd also suggest naming it `adminRedactUserMessagesIn` for consistency.

Reply:
(this also allows us to use `limit` instead of `lim`, because we wouldn't be shadowing the variable anymore)